tests/FSharp.Compiler.UnitTests/ILHelpers.fs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open System
+open System.IO
+open System.Diagnostics
+
+open NUnit.Framework
+
+module ILChecker =
+
+    let private (++) a b = Path.Combine(a,b)
+
+    let private getfullpath workDir path =
+        let rooted =
+            if Path.IsPathRooted(path) then path
+            else Path.Combine(workDir, path)
+        rooted |> Path.GetFullPath
+
+    let private fileExists workDir path = 
+        if path |> getfullpath workDir |> File.Exists then Some path else None
+
+    let private requireFile nm = 
+        if fileExists __SOURCE_DIRECTORY__ nm |> Option.isSome then nm else failwith (sprintf "couldn't find %s. Running 'build test' once might solve this issue" nm)
+
+    let exec exe args =
+        let startInfo = ProcessStartInfo(exe, String.concat " " args)
+        startInfo.RedirectStandardError <- true
+        startInfo.UseShellExecute <- false
+        use p = Process.Start(startInfo)
+        p.WaitForExit()
+        p.StandardError.ReadToEnd(), p.ExitCode
+
+    let checkContains sourceCode expectedILCode =
+        let SCRIPT_ROOT = __SOURCE_DIRECTORY__
+        let packagesDir = SCRIPT_ROOT ++ ".." ++ ".." ++ "packages"
+#if DEBUG
+        let configurationName = "debug"
+#else
+        let configurationName = "release"
+#endif
+        let fscBin = SCRIPT_ROOT ++ ".." ++ ".." ++ configurationName ++ "net40" ++ "bin"
```

Comment:
Is this the desktop fsc.exe?

Reply:
It's the compiled one when we build visualfsharp. We do this in FSharp.Tests.FSharpSuite, I just copied it over here.

Reply:
I've changed this to not directly invoke fsc, instead we just use FCS.